src/main/java/com/squareup/javapoet/MethodSpec.java

```diff
@@ -164,12 +170,27 @@ public static Builder constructorBuilder() {
   }
 
   /**
-   * Create a builder which overrides {@code method}. This will copy its visibility modifiers, type
-   * parameters, return type, name, parameters, and throws declarations. An {@link Override}
-   * annotation will be added.
+   * Create a method spec builder which overrides {@code method}.
+   *
+   * Same as {@code overriding(method, (DeclaredType) method.getEnclosingElement().asType())}.
+   */
+  public static MethodSpec.Builder overriding(ExecutableElement method,
+      Elements elements, Types types) {
+    DeclaredType containing = (DeclaredType) method.getEnclosingElement().asType();
+    return overriding(method, containing, elements, types);
+  }
+
+  /**
+   * Create a method spec builder which overrides {@code method} that is viewed as being a member
+   * of the specified {@code containing} class or interface.
+   *
+   * This will copy its visibility modifiers, type parameters, return type, name, parameters, and
+   * throws declarations. An {@link Override} annotation will be added.
    */
-  public static Builder overriding(ExecutableElement method) {
```

Comment:
This method is replaced by the one taking additional 'Elements' and 'Types' parameters around line 177. Existing code will break - but it relied on a not working/half-implemented version of `overriding` anyway.

If you know a solution without using the modell utility classes and achiving generic parameter substition, I'd be happy to learn about that. See `overrideExtendsOthersWorksWithActualTypeParameters` test.